test.js

```diff
@@ -436,3 +436,447 @@ test('highlight()', function (t) {
+      '<span class="hljs-built_in">console</span>.log(<span class="hljs-string">"Hello, "</span> + name + <span class="hljs-string">"!"</span>)</code></pre>'
+    ].join('\n'),
+    'should reprocess exact'
+  )
+
+  t.equal(
+    rehype()
+      .data('settings', {fragment: true})
+      .use(light, {
+        aliases: {tex: ['latex']},
+        languages: {latex: lt}
+      })
+      .processSync(
+        [
+          '<pre><code class="lang-latex">\\begin{document}',
+          '\\end{document}</code></pre>'
+        ].join('\n')
+      )
+      .toString(),
+    [
+      '<pre><code class="hljs lang-latex"><span class="hljs-keyword">\\begin</span>{document}',
+      '<span class="hljs-keyword">\\end</span>{document}</code></pre>'
+    ].join('\n'),
+    'should parse custom language'
+  )
+
+  t.equal(
+    rehype()
+      .data('settings', {fragment: true})
+      .use(light)
+      .processSync(
+        [
+          '<h1>Hello World!</h1>',
+          '',
+          '<pre><code class="hljs lang-js"><span class="hljs-keyword">var</span> name = <span class="hljs-string">"World"</span>;',
+          '<span class="hljs-built_in">console</span>.log(<span class="hljs-string">"Hello, "</span> + name + <span class="hljs-string">"!"</span>)</code></pre>'
+        ].join('\n')
+      )
+      .toString(),
+    [
+      '<h1>Hello World!</h1>',
+      '',
+      '<pre><code class="hljs lang-js"><span class="hljs-keyword">var</span> name = <span class="hljs-string">"World"</span>;',
+      '<span class="hljs-built_in">console</span>.log(<span class="hljs-string">"Hello, "</span> + name + <span class="hljs-string">"!"</span>)</code></pre>'
+    ].join('\n'),
+    'should reprocess exact'
+  )
+
+  t.equal(
+    rehype()
+      .data('settings', {fragment: true})
+      .use(light)
+      .processSync(
+        [
+          '<h1>Hello World!</h1>',
+          '',
+          '<pre><code><!--TODO-->"use strict";</code></pre>'
+        ].join('\n')
+      )
+      .toString(),
+    [
+      '<h1>Hello World!</h1>',
+      '',
+      '<pre><code class="hljs language-javascript"><span class="hljs-meta">"use strict"</span>;</code></pre>'
+    ].join('\n'),
+    'should ignore comments'
+  )
+
+  t.equal(
+    rehype()
+      .data('settings', {fragment: true})
+      .use(light)
+      .processSync(
+        [
+          '<h1>Hello World!</h1>',
+          '',
+          '<pre><code>"use strict";<br>console.log("very strict")</code></pre>'
+        ].join('\n')
+      )
+      .toString(),
+    [
+      '<h1>Hello World!</h1>',
+      '',
+      '<pre><code class="hljs language-javascript"><span class="hljs-meta">"use strict"</span>;',
+      '<span class="hljs-built_in">console</span>.log(<span class="hljs-string">"very strict"</span>)</code></pre>'
+    ].join('\n'),
+    'should support `<br>` elements'
+  )
+
+  t.equal(
+    rehype()
+      .data('settings', {fragment: true})
+      .use(light, {
+        languages: {
+          subunit: su,
```

Comment:
I'm cheating here! I have no clue why it generates `language-subunit` for the original test also.